DM_2/users/users_controller.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+const DefaultError = require('../exceptions/defaultError');
+const UsersModel = require('./users_model');
+const ViewsClass = require('../views/view_class');
+
+module.exports = class UsersController {
+  constructor(req, res) {
+    this.res = res;
+    this.req = req;
+    this.userModel = new UsersModel(req);
+    this.view = new ViewsClass(res);
+  }
+
+  async authUserController() {
+    try {
+      const authInfo = await this.userModel.checkIsUser();
+      if (authInfo) {
+        await this.view.setResLocalsData(this.req, this.res, true);
+      } else {
+        await this.view.setResLocalsData(this.req, this.res, false);
+        await this.view.errorData(this.res, 401,{phone: this.req.headers.userphone}, 'User is not registered');
```

Comment:
here should be used next(Error)

Reply:
used:
... 
throw this.view.errorData(this.res, 401,{phone: this.req.headers.userphone}, 'User is not registered');
catch(err) {
next(err)
}